src/bag.test.js

```diff
@@ -183,6 +183,42 @@ describe("rosbag - high-level api", () => {
       topics.forEach((topic) => expect(topic).toBe("/turtle1/color_sensor"));
     });
 
+    it("asynchronously reads bz2 with supplied decompression callback", async () => {
+      const filename = getFixture("example-bz2");
+      const bag = await Bag.open(filename);
+
+      const messages = [];
+      await bag.readMessages({
+        topics: ["/turtle1/color_sensor"],
+        decompress: {
+          bz2: (buffer: Buffer) => {
+            return new Promise((resolve) => {
+              setTimeout(() => {
+                const arr = compress.Bzip2.decompressFile(buffer);
+                resolve(Buffer.from(arr));
+              }, 100);
+            });
+          },
+        },
+      }, (msg) => messages.push(msg));
+
+      const topics = messages.map((msg) => msg.topic);
+      expect(topics).toHaveLength(1351);
+      topics.forEach((topic) => expect(topic).toBe("/turtle1/color_sensor"));
+
+      for (let i = 1, l = messages.length; i < l; i ++) {
```

Comment:
These last expects aren't all that useful with a bag that just has a single chunk in it.